src/compiler/program.ts

```diff
@@ -1804,25 +1804,26 @@ namespace ts {
 
         function getDiagnosticsHelper<T extends Diagnostic>(
             sourceFile: SourceFile | undefined,
-            getDiagnostics: (sourceFile: SourceFile, cancellationToken: CancellationToken | undefined) => readonly T[],
-            cancellationToken: CancellationToken | undefined): readonly T[] {
+            getDiagnostics: (sourceFile: SourceFile, cancellationToken: CancellationToken | undefined, includeUncheckedJS: boolean | undefined) => readonly T[],
+            cancellationToken: CancellationToken | undefined,
+            includeUncheckedJS: boolean | undefined): readonly T[] {

```

Comment:
Why is it preferable to add this to getDiagnosticsHelper instead of just passing it through to getSemanticDiagnosticsForFile in getSemanticDiagnostics?

Reply:
No good reason, I was just following compile errors around and mechanically adding parameters.